panel/applets/sensors/SensorsApplet.vala

```diff
@@ -0,0 +1,210 @@
+                  });
+
+          getSensors();
+          add(m_widget);
+          show_all();
+
+          m_running = true;
+          Timeout.add_seconds_full(GLib.Priority.LOW, 1, getSensors);
+  }
+
+  public override void update_popovers(Budgie.PopoverManager ? manager)
+  {
+          m_manager = manager;
+          m_manager.register_popover(m_widget, m_popover);
+  }
+
+  void on_settings_change(string key)
+  {
+          if ("unit" == key)
+          {
+                  int unit = m_settings.get_int(key);
+                  switch (unit)
+                  {
+                  case 0 :
+                          m_isCelsius = true;
+                          break;
+
+                  case 1 :
+                          m_isCelsius = false;
+                          break;
+                  }
+          }
+  }
+
+  public void create_menuMain()
+  {
+          m_menuMain = new GLib.Menu();
+          m_menuMain.append ("Celsius", "celsius");
+          m_menuMain.append ("Fahrenheit", "fahrenheit");
+
+          var celsius_action = new SimpleAction ("celsius", null);
+          celsius_action.activate.connect (this.celsius_cb);
+
+          var fahrenheit_action = new SimpleAction ("fahrenheit", null);
+          fahrenheit_action.activate.connect (this.fahrenheit_cb);
+  }
+
+
+  void celsius_cb (SimpleAction simple, Variant ? parameter) {
+          m_isCelsius = true;
+          m_settings.set_int ("unit", 0);
+  }
+
+  void fahrenheit_cb (SimpleAction simple, Variant ? parameter) {
+          m_isCelsius = false;
+          m_settings.set_int ("unit", 1);
+  }
+
+  private bool getSensors()
+  {
+          string ls_stdout;
+          string ls_stderr;
+          int ls_status;
+          string editStr = "";
+
+          string [] lines;
+          string [] elines;
+          string [] e2lines;
+
+          lock (m_lock)
+          {
+                  if (m_isCelsius == true)
+                  {
+                          m_exec = Process.spawn_command_line_sync ("sensors",
+                                                                    out ls_stdout,
+                                                                    out ls_stderr,
+                                                                    out ls_status);
+                  }
+                  else
+                  {
+                          m_exec = Process.spawn_command_line_sync ("sensors -f",
+                                                                    out ls_stdout,
+                                                                    out ls_stderr,
+                                                                    out ls_status);
+                  }
+                  if (ls_stdout != null)
+                  {
+                          lines = ls_stdout.split("\n");
+                          for (int i =0; i < lines.length; i++)
+                          {
+                                  if ("Core" in lines[i])
```

Comment:
JS Style equality comparison, you're comparing a current _thing_ to a constant, so the constant should be the Y param